lib/tokenize.cpp

```diff
@@ -9414,11 +9416,18 @@ void Tokenizer::simplifyCPPAttribute()
                 if (head && head->str() == "(" && isFunctionHead(head, "{|;")) {
                     head->previous()->isAttributeNodiscard(true);
                 }
-            } else if (Token::findsimplematch(tok->tokAt(2), "maybe_unused", tok->link())) {
+            } else if ((Token::findsimplematch(tok->tokAt(2), "maybe_unused", tok->link()) && hasMaybeUnused)
```

Comment:
If you write this code: `func() && flag` the compiler is required to execute func() even if it knows that flag is false. And.. it is not allowed to optimise and generate the code `flag && func()` . 

Please put the flag first.